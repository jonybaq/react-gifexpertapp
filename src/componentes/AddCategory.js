import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {
    const [texto,SetTexto]=useState('');
    const handleChange=(e)=>{ SetTexto(e.target.value);  }
    const handleSubmit=(e)=>{
        e.preventDefault();
        if (texto.trim().length>2) {
            setCategories(cat=>[texto,...cat]);
        }
        
        SetTexto("");
    }
    return (
        <form onSubmit={handleSubmit}>
            <h2>AÃ±adir Categoria</h2>
            <h4>{texto}</h4>
            <input type="text" value={texto} placeholder='Nueva Categoria' onChange={handleChange} />
        </form>
    )
}
AddCategory.propTypes={
    setCategories: PropTypes.func.isRequired
}
