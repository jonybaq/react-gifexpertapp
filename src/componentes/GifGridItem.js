import React from 'react'

export const GifGridItem = ({id,title='(Sin Titulo)',url}) => {
    //console.log(id);

        if (title.length===0) {
            title="(Sin Titulo)";
        }

    return (
        <div className="card animate__animated animate__bounce animate__delay-2s">
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}
