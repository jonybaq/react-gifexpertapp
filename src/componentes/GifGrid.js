import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({categoria}) => {
    const [images, setImages] = useState([])
    useEffect(() => {
        const getGifs=async()=>{
            const api_key="3oqN5d8hX8sWktmbdSEOr7xDcT83uTRP";
            const url="http://api.giphy.com/v1/gifs/search?api_key="+api_key+"&q="+categoria+"&limit=10";
            const res=await fetch(url);
            const {data}= await res.json();
            //console.log(data);
            const gifs=data.map(img=>{
                return {
                    id: img.id,
                    title:img.title,
                    url:img.images?.downsized_medium.url
                }
            });
            //console.log(gifs);
            setImages(gifs);
        }
        getGifs();
    },[categoria]);
    
    return (
        <>
        <h3>{categoria}</h3>
        <div className="card-grid"> 
            {
                images.map((image)=>
                    <GifGridItem key={image.id}{...image}/>   
                )
            }
            
        </div>
        </>
    )
}
