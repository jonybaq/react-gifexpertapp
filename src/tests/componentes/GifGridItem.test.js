
import { shallow } from "enzyme"
import React from "react";
import {GifGridItem} from '../../componentes/GifGridItem';
describe('Prueba del GifGriditem.test.js', () => {
    test('DEBE DE MOSTRAR EL COMPONENTE CORRECTAMENTE', () => {
        const wrapper=shallow(<GifGridItem/>);
        expect(wrapper).toMatchSnapshot();
    })
    
})
