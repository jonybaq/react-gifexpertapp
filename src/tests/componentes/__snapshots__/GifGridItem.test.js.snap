// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del GifGriditem.test.js DEBE DE MOSTRAR EL COMPONENTE CORRECTAMENTE 1`] = `
<div
  className="card animate__animated animate__bounce animate__delay-2s"
>
  <img
    alt="(Sin Titulo)"
  />
  <p>
    (Sin Titulo)
  </p>
</div>
`;
